operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/DependentResource.java

```diff
@@ -0,0 +1,15 @@
+package io.javaoperatorsdk.operator.api.config;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.javaoperatorsdk.operator.api.reconciler.EventSourceContext;
+import io.javaoperatorsdk.operator.processing.event.source.EventSource;
+
+public interface DependentResource<R, P extends HasMetadata> {
+  default EventSource initEventSource(EventSourceContext<P> context) {
+    throw new IllegalStateException("Must be implemented if not automatically provided by the SDK");
```

Comment:
This is debatable, indeed, though I thought there was a reason for this at some point but don't recall why at the moment.

Reply:
maybe we could remove it now, and if we find a reason we put it back and describe it in a comment

Reply:
OK, I've checked and it's actually done this way because this allows dependent definitions for Kubernetes resources to not have to provide an implementation because one is provided automatically in that case. However, for external resources, the user needs to provide one so the solution was to throw the exception if an implementation is not automatically provided by the framework.